patches/minecraft/net/minecraft/block/ConcretePowderBlock.java.patch

```diff
@@ -0,0 +1,60 @@
++++ b/net/minecraft/block/ConcretePowderBlock.java
+@@ -20,8 +20,9 @@
+    }
+ 
+    public void func_176502_a_(World p_176502_1_, BlockPos p_176502_2_, BlockState p_176502_3_, BlockState p_176502_4_, FallingBlockEntity p_176502_5_) {
+-      if (func_230137_b_(p_176502_1_, p_176502_2_, p_176502_4_)) {
+-         p_176502_1_.func_180501_a(p_176502_2_, this.field_200294_a, 3);
++      net.minecraftforge.event.world.BlockEvent.ConcretePowderConversionEvent event = net.minecraftforge.event.ForgeEventFactory.onConvertConcretePowder(p_176502_1_, p_176502_2_, p_176502_4_, this.field_200294_a);
++      if (event.getResult() != net.minecraftforge.eventbus.api.Event.Result.DENY && (event.getResult() == net.minecraftforge.eventbus.api.Event.Result.ALLOW || (event.getResult() == net.minecraftforge.eventbus.api.Event.Result.DEFAULT && func_230137_b_(p_176502_1_, p_176502_2_, p_176502_4_)))) {
++         p_176502_1_.func_180501_a(p_176502_2_, event.getConvertedState(), 3);
+       }
+ 
+    }
+@@ -30,11 +31,12 @@
+       IBlockReader iblockreader = p_196258_1_.func_195991_k();
+       BlockPos blockpos = p_196258_1_.func_195995_a();
+       BlockState blockstate = iblockreader.func_180495_p(blockpos);
+-      return func_230137_b_(iblockreader, blockpos, blockstate) ? this.field_200294_a : super.func_196258_a(p_196258_1_);
++      net.minecraftforge.event.world.BlockEvent.ConcretePowderConversionEvent event = net.minecraftforge.event.ForgeEventFactory.onConvertConcretePowder(p_196258_1_.func_195991_k(), blockpos, blockstate, this.field_200294_a);
++      return event.getResult() != net.minecraftforge.eventbus.api.Event.Result.DENY && (event.getResult() == net.minecraftforge.eventbus.api.Event.Result.ALLOW || (event.getResult() == net.minecraftforge.eventbus.api.Event.Result.DEFAULT && func_230137_b_(iblockreader, blockpos, blockstate))) ? event.getConvertedState() : super.func_196258_a(p_196258_1_);
+    }
+ 
+    private static boolean func_230137_b_(IBlockReader p_230137_0_, BlockPos p_230137_1_, BlockState p_230137_2_) {
+-      return func_212566_x(p_230137_2_) || func_196441_b(p_230137_0_, p_230137_1_);
++      return causesSolidify(p_230137_2_, null, p_230137_0_, p_230137_1_) || func_196441_b(p_230137_0_, p_230137_1_);
+    }
+ 
+    private static boolean func_196441_b(IBlockReader p_196441_0_, BlockPos p_196441_1_) {
+@@ -43,25 +45,24 @@
+ 
+       for(Direction direction : Direction.values()) {
+          BlockState blockstate = p_196441_0_.func_180495_p(blockpos$mutable);
+-         if (direction != Direction.DOWN || func_212566_x(blockstate)) {
+             blockpos$mutable.func_239622_a_(p_196441_1_, direction);
+             blockstate = p_196441_0_.func_180495_p(blockpos$mutable);
+-            if (func_212566_x(blockstate) && !blockstate.func_224755_d(p_196441_0_, p_196441_1_, direction.func_176734_d())) {
++            if (causesSolidify(blockstate, direction.func_176734_d(), p_196441_0_, blockpos$mutable)) {
```

Comment:
Why does suddenly only check the method? And not the event? You have to be consistent.